rafx-api/src/backends/gles2/internal/conversions.rs

```diff
@@ -0,0 +1,242 @@
+            RafxAddressMode::ClampToBorder => None,
+        }
+    }
+}
+
+impl RafxPrimitiveTopology {
+    pub fn gles2_topology(self) -> Option<GLenum> {
+        match self {
+            RafxPrimitiveTopology::PointList => Some(gles2_bindings::POINTS),
+            RafxPrimitiveTopology::LineList => Some(gles2_bindings::LINES),
+            RafxPrimitiveTopology::LineStrip => Some(gles2_bindings::LINE_STRIP),
+            RafxPrimitiveTopology::TriangleList => Some(gles2_bindings::TRIANGLES),
+            RafxPrimitiveTopology::TriangleStrip => Some(gles2_bindings::TRIANGLE_STRIP),
+            RafxPrimitiveTopology::PatchList => None,
+        }
+    }
+}
+
+impl RafxMemoryUsage {
+    pub fn gles2_usage(self) -> Option<GLenum> {
+        match self {
+            RafxMemoryUsage::Unknown => None,
+            RafxMemoryUsage::GpuOnly => Some(gles2_bindings::STATIC_DRAW),
+            RafxMemoryUsage::CpuOnly => Some(gles2_bindings::NONE),
+            RafxMemoryUsage::CpuToGpu => Some(gles2_bindings::DYNAMIC_DRAW),
+            RafxMemoryUsage::GpuToCpu => Some(gles2_bindings::STREAM_DRAW),
+        }
+    }
+}
+
+impl RafxCullMode {
+    pub fn gles2_cull_mode(self) -> GLenum {
+        match self {
+            RafxCullMode::None => gles2_bindings::NONE,
+            RafxCullMode::Back => gles2_bindings::BACK,
+            RafxCullMode::Front => gles2_bindings::FRONT,
+        }
+    }
+}
+
+impl RafxFrontFace {
+    pub fn gles2_front_face(self) -> GLenum {
+        match self {
+            RafxFrontFace::CounterClockwise => gles2_bindings::CCW,
+            RafxFrontFace::Clockwise => gles2_bindings::CW,
+        }
+    }
+}
+
+impl RafxCompareOp {
+    pub fn gles2_compare_op(self) -> GLenum {
+        match self {
+            RafxCompareOp::Never => gles2_bindings::NEVER,
+            RafxCompareOp::Less => gles2_bindings::LESS,
+            RafxCompareOp::Equal => gles2_bindings::EQUAL,
+            RafxCompareOp::LessOrEqual => gles2_bindings::LEQUAL,
+            RafxCompareOp::Greater => gles2_bindings::GREATER,
+            RafxCompareOp::NotEqual => gles2_bindings::NOTEQUAL,
+            RafxCompareOp::GreaterOrEqual => gles2_bindings::GEQUAL,
+            RafxCompareOp::Always => gles2_bindings::ALWAYS,
+        }
+    }
+}
+
+impl RafxStencilOp {
+    pub fn gles2_stencil_op(self) -> GLenum {
+        match self {
+            RafxStencilOp::Keep => gles2_bindings::KEEP,
+            RafxStencilOp::Zero => gles2_bindings::ZERO,
+            RafxStencilOp::Replace => gles2_bindings::REPLACE,
+            RafxStencilOp::IncrementAndClamp => gles2_bindings::INCR,
+            RafxStencilOp::DecrementAndClamp => gles2_bindings::DECR,
+            RafxStencilOp::Invert => gles2_bindings::INVERT,
+            RafxStencilOp::IncrementAndWrap => gles2_bindings::INCR_WRAP,
+            RafxStencilOp::DecrementAndWrap => gles2_bindings::INCR_WRAP,
+        }
+    }
+}
+
+//TODO: Some fields in RafxRasterizerState are not handled!
```

Comment:
TODO seems relevant, ditto for the one below on `RafxDepthState`.

Reply:
I took a quick glance, the color masks were easy to implement but the others will take more time than I want to spend right now